references/classification/train.py

```diff
@@ -154,18 +148,13 @@ def load_data(traindir, valdir, args):
         print(f"Loading dataset_test from {cache_path}")
         dataset_test, _ = torch.load(cache_path)
     else:
-        if not args.prototype:
+        if args.weights and args.test_only:
+            weights = torchvision.models.get_weight(args.weights)
+            preprocessing = weights.transforms()
```

Comment:
We are going to use the bundled transforms only when we load weights and validate a model. When training happens (either full training or fine-tuning) we always want to use the presets defined in our reference scripts.

Reply:
Documenting this, as we just discussed offline:

- we want users to be able to use the builtin presets (`weights.transforms()`) for inference
- yet we still want users to be able to modify the presets they're using on the references for their own experiments. So it makes sense to have somewhat duplicated presets, the builtin ones and the ones on the references.
- we don't have a great weight to test the builtin presets right now, since we don't test model training (but we're planning to work on this). So relying on the builtin presets here acts somewhat as a functional test for them. It's also relevant to rely on them in `test_only` mode, because these are the actual presets that we except to be used in production settings.